store/postgres/src/entity_changes.rs

```diff
@@ -65,53 +62,46 @@ impl EntityChangeListener {
-                        // Parse payload into an entity change
-                        let value: serde_json::Value =
-                            serde_json::from_str(notification.payload.as_str())
-                                .expect("Invalid JSON entity change data received from database");
-                        let change: EntityChange = serde_json::from_value(value.clone()).expect(
-                            format!(
-                                "Invalid entity change received from the database: {:?}",
-                                value
-                            ).as_str(),
-                        );
-
-                        // We'll assume here that if sending fails, this means that the
-                        // entity change listener has already been dropped, the receiving
-                        // is gone and we should terminate the listener loop
-                        if sender.clone().send(change).wait().is_err() {
-                            break;
-                        }
+            // Read notifications until the thread is to be terminated
+            while !terminate.load(Ordering::SeqCst) {
+                // Obtain a notifications iterator from Postgres
+                let notifications = conn.notifications();
+
+                // Read notifications until there hasn't been one for 500ms
+                for notification in notifications
+                    .timeout_iter(Duration::from_millis(500))
+                    .iterator()
+                    .filter_map(Result::ok)
+                    .filter(|notification| notification.channel == "entity_changes")
+                {
+                    // Terminate the thread if desired
+                    if terminate.load(Ordering::SeqCst) {
+                        return;
                     }
-                    Err(e) => {
-                        error!(logger, "Failed to retrieve notification from Postgres";
-                           "error" => format!("{}", e).as_str());
+
+                    // Parse payload into an entity change
+                    let value: serde_json::Value =
+                        serde_json::from_str(notification.payload.as_str())
+                            .expect("Invalid JSON entity change data received from database");
+                    let change: EntityChange = serde_json::from_value(value.clone()).expect(
+                        format!(
+                            "Invalid entity change received from the database: {:?}",
+                            value
+                        ).as_str(),
+                    );
+
+                    // We'll assume here that if sending fails, this means that the
+                    // entity change listener has already been dropped, the receiving
+                    // is gone and we should terminate the listener loop
+                    if sender.clone().send(change).wait().is_err() {
+                        break;
```

Comment:
But you're right, this break looks strange. Since we now set `terminate` upon dropping the `EntityChangeListener`, then this is redundant since we'll break out anyways at the `while` check.

Reply:
I think we still want to `break` here to avoid reading any more notifications. But `break`ing is enough after all, since the terminate check will kick in right away.